internal/services/securitycenter/security_center_server_vulnerability_assessment_virtual_machine_resource_test.go

```diff
@@ -0,0 +1,141 @@
+package securitycenter_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type ServerVulnerabilityAssessmentVirtualMachineResource struct{}
+
+func TestAccServerVulnerabilityAssessmentVirtualMachine_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_security_center_server_vulnerability_assessment_virtual_machine", "test")
+	r := ServerVulnerabilityAssessmentVirtualMachineResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basicCfg(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccServerVulnerabilityAssessmentVirtualMachine_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_security_center_server_vulnerability_assessment_virtual_machine", "test")
+	r := ServerVulnerabilityAssessmentVirtualMachineResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basicCfg(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (ServerVulnerabilityAssessmentVirtualMachineResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := azure.ParseAzureResourceID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceGroup := id.ResourceGroup
+	name := id.Path["virtualMachines"]
```

Comment:
this should be using the `parse.VulnerabilityAssessmentVmID(state.ID)` method:

```suggestion
	id, err := parse.VulnerabilityAssessmentVmID(state.ID)
	if err != nil {
		return nil, err
	}
```

we'll be removing `azure.ParseAzureResourceID` in future fwiw